runtime/commonTest/src/kotlinx/serialization/json/JsonMapKeysTest.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json
+
+import kotlinx.serialization.*
+import kotlinx.serialization.test.assertStringFormAndRestored
+import kotlin.test.Test
+import kotlin.test.assertFailsWith
+
+@Serializable
+private data class WithMap(val map: Map<Long, Long>)
+
+@Serializable
+private data class WithEnum(val map: Map<SampleEnum, Long>)
+
+@Serializable
+private data class WithComplexKey(val map: Map<IntData, String>)
+
+class JsonMapKeysTest : JsonTestBase() {
```

Comment:
Now polymorphic serialization with complex keys should work out of the box (#480).
Please uncomment `JsonMapPolymorphismTest.testPolymorphicKeys` and add more tests for that